.github/workflows/aws-e2e.yaml

```diff
@@ -0,0 +1,121 @@
+'on':
+  push:
+    branches:
+    - main
+  pull_request:
```

Comment:
wdyt, if we just run it on pull_request until we completely migrate to TC?
and disable all TC configurations for e2e tests. just Keep TC configurations that runs on main and release which actually builds AMI without `--dry-run`.

Reply:
changed to just pull requests and will do the runs without `--dry-run` when refactoring to a shared workflow